cloud: Premium supports data migration (#22821)
pingcap/docs: tidb-cloud/migrate-from-mysql-using-data-migration.md

Reviewer: ```suggestion
<CustomContent plan="dedicated,premium">
```

Reviewer: ```suggestion
<CustomContent plan="dedicated,premium">
```

</CustomContent>

<CustomContent plan="premium">

> - For {{{ .premium }}}, both logical mode (default) and physical mode are supported. Logical mode exports data from MySQL source databases as SQL statements and then executes them on the target {{{ .premium }}} instance, which consumes Request Capacity Units (RCUs) during the load. Physical mode uses `IMPORT INTO` on the target {{{ .premium }}} instance and is recommended for large datasets when load throughput and cost are priorities.

Reviewer 2: ```suggestion
- For {{{ .premium }}}, both logical mode (default) and physical mode are supported. Logical mode exports data from MySQL source databases as SQL statements and then executes them on the target {{{ .premium }}} instance, which consumes Request Capacity Units (RCUs) during the load. Physical mode uses `IMPORT INTO` on the target {{{ .premium }}} instance and is recommended for large datasets when you prioritize load throughput and cost efficiency.
```

</CustomContent>

<CustomContent plan="premium">

> For {{{ .premium }}}, the Data Migration feature supports any MySQL-compatible source database, and **MySQL** is the only data source type available in the migration job wizard. For supported connection methods, see [Ensure network connectivity](#ensure-network-connectivity).

Reviewer 2: ```suggestion
For {{{ .premium }}}, the Data Migration feature supports the following MySQL-compatible source databases, and **MySQL** is the only data source type available in the migration job wizard. For supported connection methods, see [Ensure network connectivity](#ensure-network-connectivity).
```

    - **Supported IP address types**: select **IPv4**.

    After the endpoint service is created, copy the service name for later use. The service name is in the `com.amazonaws.vpce.<region>.vpce-svc-<id>` format, for example, `com.amazonaws.vpce.us-east-1.vpce-svc-0123456789abcdef0`.

    For detailed instructions, see [Create an endpoint service](https://docs.aws.amazon.com/vpc/latest/privatelink/create-endpoint-service.html) in AWS documentation.

3. Authorize TiDB Cloud's AWS principal to use your endpoint service. On the endpoint service detail page in the [Amazon VPC console](https://console.aws.amazon.com/vpc/), open the **Allow principals** tab, click **Allow principals**, and add the following ARN:

    ```text
    arn:aws:iam::886436925895:root
    ```

    Without this step, TiDB Cloud cannot create the VPC endpoint that connects to your service, and the **Create Private Endpoint for External Services** dialog in TiDB Cloud hangs indefinitely with no error.

    For detailed instructions, see [Manage permissions](https://docs.aws.amazon.com/vpc/latest/privatelink/configure-endpoint-service.html#add-remove-permissions) in AWS documentation.

> 4. Optional: Test connectivity from a bastion or client inside the same VPC or VNet before starting the migration:

Reviewer 2: ```suggestion
4. Optional: Test connectivity from a bastion or client inside the same VPC before starting the migration:
```

Reviewer 2: Reason: “VNet” is an Azure term, but this step is for AWS only.

    - **Supported IP address types**: select **IPv4**.

    After the endpoint service is created, copy the service name for later use. The service name is in the `com.amazonaws.vpce.<region>.vpce-svc-<id>` format, for example, `com.amazonaws.vpce.us-east-1.vpce-svc-0123456789abcdef0`.

    For detailed instructions, see [Create an endpoint service](https://docs.aws.amazon.com/vpc/latest/privatelink/create-endpoint-service.html) in AWS documentation.

3. Authorize TiDB Cloud's AWS principal to use your endpoint service. On the endpoint service detail page in the [Amazon VPC console](https://console.aws.amazon.com/vpc/), open the **Allow principals** tab, click **Allow principals**, and add the following ARN:

    ```text
    arn:aws:iam::886436925895:root
    ```

    Without this step, TiDB Cloud cannot create the VPC endpoint that connects to your service, and the **Create Private Endpoint for External Services** dialog in TiDB Cloud hangs indefinitely with no error.

    For detailed instructions, see [Manage permissions](https://docs.aws.amazon.com/vpc/latest/privatelink/configure-endpoint-service.html#add-remove-permissions) in AWS documentation.

> 4. Optional: Test connectivity from a bastion or client inside the same VPC or VNet before starting the migration:

Reviewer 2: ```suggestion
4. Optional: Test connectivity from a bastion or client inside the same VPC before starting the migration:
```

Reviewer 2: Reason: “VNet” is an Azure term, but this step is for AWS only.

If your MySQL service is in an AWS VPC, take the following steps:

> 1. [Set up a VPC peering connection](/tidb-cloud/set-up-vpc-peering-connections.md) between the VPC of the MySQL service and your <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="essential">{{{ .essential }}} instance</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent>.

Reviewer 2: ```suggestion
1. [Set up a VPC peering connection](/tidb-cloud/set-up-vpc-peering-connections.md) between the VPC of the MySQL service and your <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent>.
```

Reviewer 2: As in line 552, this section is for Dedicated cluster only.

If your MySQL service is in an AWS VPC, take the following steps:

> 1. [Set up a VPC peering connection](/tidb-cloud/set-up-vpc-peering-connections.md) between the VPC of the MySQL service and your <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent>.

Reviewer 2: ```suggestion
1. [Set up a VPC peering connection](/tidb-cloud/set-up-vpc-peering-connections.md) between the VPC of the MySQL service and your {{{ .dedicated }}} cluster.
```

GRANT SELECT, RELOAD, LOCK TABLES, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO 'dm_source_user'@'%';
```

> #### Grant required privileges in the target <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="essential">{{{ .essential }}} instance</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent>

Reviewer 2: ```suggestion
#### Grant required privileges in the target TiDB Cloud resource
```

> **Note:**
>
> - When you use physical mode, you cannot create a second migration job or import task for the <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="essential">{{{ .essential }}} instance</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent> before the existing data migration is completed.

> > - When you use physical mode and the migration job has started, do **NOT** enable PITR (Point-in-time Recovery) or have any changefeed on the <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="essential">{{{ .essential }}} instance</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent>. Otherwise, the migration job stops. If you need to enable PITR or have any changefeed, use logical mode instead to migrate data.

Reviewer 2: As in line 812, the two notes above belong to `<CustomContent plan="dedicated">`, so we might need to remove Essential and Premium content from it and consider whether to add additional content for Premium.

> > - When you use physical mode and the migration job has started, do **NOT** enable PITR (Point-in-time Recovery) or have any changefeed on the <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="essential">{{{ .essential }}} instance</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent>. Otherwise, the migration job stops. If you need to enable PITR or have any changefeed, use logical mode instead to migrate data.

Reviewer: ```suggestion
> - When you use physical mode, you cannot create a second migration job or import task for the <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent> before the existing data migration is completed.
> - When you use physical mode and the migration job has started, do **NOT** enable PITR (Point-in-time Recovery) or have any changefeed on the <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent>. Otherwise, the migration job stops. If you need to enable PITR or have any changefeed, use logical mode instead to migrate data.
```

1. Log in to the [TiDB Cloud console](https://tidbcloud.com/) and navigate to the [**My TiDB**](https://tidbcloud.com/tidbs) page.

> 2. Click the name of your target <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="essential">{{{ .essential }}} instance</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent> to go to its overview page, and then click **Data** > **Data Migration** in the left navigation pane.

Reviewer 2: Same issue here: all content starting from line 899 belongs to `<CustomContent plan="dedicated">`, so we might need to remove Essential and Premium content from the line above and consider whether to add additional sections for Premium.

> 2. Click the name of your target <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="essential">{{{ .essential }}} instance</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent> to go to its overview page, and then click **Data** > **Data Migration** in the left navigation pane.

Reviewer: ```suggestion
2. Click the name of your target <CustomContent plan="dedicated">{{{ .dedicated }}} cluster</CustomContent><CustomContent plan="premium">{{{ .premium }}} instance</CustomContent> to go to its overview page, and then click **Data** > **Data Migration** in the left navigation pane.
```
